sdk/trace/id_generator.go

```diff
@@ -15,20 +15,26 @@
 package trace // import "go.opentelemetry.io/otel/sdk/trace"
 
 import (
+	crand "crypto/rand"
+	"encoding/binary"
 	"math/rand"
 	"sync"
 
 	"go.opentelemetry.io/otel/trace"
-
-	"go.opentelemetry.io/otel/sdk/trace/internal"
 )
 
+// IDGenerator allows custom generators for TraceId and SpanId.
+type IDGenerator interface {
+	NewTraceID() trace.TraceID
+	NewSpanID() trace.SpanID
+}
+
 type defaultIDGenerator struct {
 	sync.Mutex
 	randSource *rand.Rand
 }
 
-var _ internal.IDGenerator = &defaultIDGenerator{}
+var _ IDGenerator = &defaultIDGenerator{}
```

Comment:
How about renaming the `defaultIDGenerator` type to `randomIDGenerator`, and then renaming the `defIDGenerator` function as `defaultIDGenerator`?